docs/CODE-GUIDELINES.md

```diff
@@ -28,6 +28,29 @@ assertThat(ClassToTest.methodToTest("input"), equalTo("expected"));
 assertThat(ClassToTest.methodReturnsNull(), equalTo(null));
 ```
 
+### Backtick test naming
```

Comment:
This section is definitely all up for debate. I'd just noticed that we hadn't really found a consistent style, so wanted to propose one for discussed!